### PR TITLE
Update to python 3.11

### PR DESCRIPTION
**Pull request by bdestombe**
netcdf4 to minimum of 1.6.4. Added flox as requirement to speed up resampling.

### CODE REVIEW

**Comment by bdestombe** [comment]
#172

**Comment by BSchilperoort** [comment]
> Added flox as requirement to speed up resampling

Nice, I have been using flox a lot recently. It has some really nice features (such as a groupby_bins that allows overlapping groups)

**Comment by bdestombe** [comment]
https://github.com/Unidata/netcdf4-python/issues/1205

**Comment by BSchilperoort** [comment]
Seems like there is an issue with string type variables in our data. My opinion is that xarray should be catching that before dispatching the writing to the netcdf4 library.

Anyway, I would think that we can manually set the compression/encoding to `None` for any string variables that are in the data, which then fixes the issue?

**Comment by bdestombe** [comment]
Okay that worked. It is now stuck in to_mf_netcdf() and I can't seem to solve that one.

**Comment by BSchilperoort** [comment]
Working on a patch, I'll push to this branch

**Comment by bdestombe** [comment]
The problem was that on loaded dask arrays, strings are loaded as datatype objects and therefore were not identified as strings. As a result those datatype objects were also getting a zlib=True in their encodings

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,10 +7,13 @@ dev
 New features
 
 * Improved the functionality of `merge_double_ended`, by adding a check that handles measurements missing in one channel while present in the other ([#171](https://github.com/dtscalibration/python-dts-calibration/pull/171))
+* Support for Python 3.11
+* Requiring netcdf4 >= 1.6.4
```
Comment (BSchilperoort) [inline CHANGELOG.rst:11]: Not really a new feature I'd think?

I think I like such a layout more for changelogs:

**Fixed:**
 - Bug in abc

**Added:**
 - Support for Python 3.11

**Changed:**
 - Now requires `netcdf >= 1.6.4`

**Review by BSchilperoort — APPROVED** [review]
Awesome! Good work finding out the dask type issue

**Pull request merged**